### PR TITLE
chore(fix): directpath ipv4 connection should not fail for an ipv6-enabled test client

### PR DESCRIPTION
**Pull request by WeiranFang**
With the new DNS support, a client running on ipv6-enabled VM could receive both ipv4 and ipv6 balancer addresses. And client could either pick the ipv4 address or the ipv6 address, and different languages of grpc have different behaviors. For Java, it will pick ipv4 address by default, so we should allow that connection when running directpath-ipv6 tests.

### CODE REVIEW

**Comment by WeiranFang** [comment]
@igorbernstein2 Hi Igor would you be able to take a look at this PR?

File: google-cloud-bigtable/pom.xml [hunk 1]
```diff
@@ -359,16 +359,16 @@
                     <bigtable.env>cloud</bigtable.env>
                     <bigtable.data-endpoint>${bigtable.directpath-data-endpoint}</bigtable.data-endpoint>
                     <bigtable.admin-endpoint>${bigtable.directpath-admin-endpoint}</bigtable.admin-endpoint>
-                    <bigtable.grpc-log-dir>${project.build.directory}/test-grpc-logs/directpath-ipv4-it</bigtable.grpc-log-dir>
+                    <bigtable.grpc-log-dir>${project.build.directory}/test-grpc-logs/directpath-ipv4only-it</bigtable.grpc-log-dir>
                     <bigtable.attempt-directpath>true</bigtable.attempt-directpath>
                     <bigtable.directpath-ipv4>true</bigtable.directpath-ipv4>
```
Comment (igorbernstein2) [inline google-cloud-bigtable/pom.xml:364]: bigtable.directpath-ipv4only

File: google-cloud-bigtable/src/test/java/com/google/cloud/bigtable/test_helpers/env/CloudEnv.java [hunk 2]
```diff
@@ -241,7 +241,8 @@ private boolean verifyRemoteAddress(SocketAddress remoteAddr) {
       if (isDirectPathIpv4()) {
```
Comment (igorbernstein2) [inline google-cloud-bigtable/src/test/java/com/google/cloud/bigtable/test_helpers/env/CloudEnv.java:241]: isDirectPathIpv4Only()

File: google-cloud-bigtable/src/test/java/com/google/cloud/bigtable/test_helpers/env/CloudEnv.java [hunk 3]
```diff
@@ -222,7 +222,7 @@ public void onHeaders(Metadata headers) {
                                   + " to the test environment's requirement for DirectPath."
                                   + " Expected test to access DirectPath via %s,"
                                   + " but RPC was destined for %s",
-                              isDirectPathIpv4() ? "ipv4" : "ipv6", remoteAddr.toString()));
+                              isDirectPathIpv4Only() ? "ipv4" : "ipv6", remoteAddr.toString()));
```
Comment (igorbernstein2) [inline google-cloud-bigtable/src/test/java/com/google/cloud/bigtable/test_helpers/env/CloudEnv.java:225]: last nit: please fix the error message

**Pull request merged**